src/storage/txn/scheduler.rs

```diff
@@ -1104,4 +1112,49 @@ mod tests {
         scheduler.run_cmd(cmd.cmd, StorageCallback::Boolean(cb));
         assert!(block_on(f).is_ok());
     }
+
+    #[test]
+    fn test_accumulate_many_expired_commands() {
+        let engine = TestEngineBuilder::new().build().unwrap();
+        let scheduler = Scheduler::new(
+            engine,
+            DummyLockManager,
+            ConcurrencyManager::new(1.into()),
+            1024,
+            1,
+            100 * 1024 * 1024,
+            Arc::new(AtomicBool::new(true)),
+            false,
+        );
+
+        let mut lock = Lock::new(&[Key::from_raw(b"b")]);
+        let cid = scheduler.inner.gen_id();
+        assert!(scheduler.inner.latches.acquire(&mut lock, cid));
+
+        // Push lots of requests in the queue.
+        for _ in 0..65536 {
```

Comment:
Can it reproduce the stack overflow?

Reply:
Yes. Without re-spawning the task, running this test will cause a stack overflow.